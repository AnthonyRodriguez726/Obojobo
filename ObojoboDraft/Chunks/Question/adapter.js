import Common from 'Common'

const { OboModel } = Common.models

const Adapter = {
	construct(model) {
		model.setStateProp('mode', 'practice')
		model.setStateProp('solution', null, p => OboModel.create(p))
	},

	clone(model, clone) {
		clone.modelState.type = model.modelState.type
		clone.modelState.mode = model.modelState.mode
		clone.modelState.solution = null

		if (model.modelState.solution) {
			clone.modelState.solution = Object.assign({}, model.modelState.solution)
		}
	},

	toJSON(model, json) {
		json.content.type = model.modelState.type
		json.content.mode = model.modelState.mode
		json.content.solution = null

		if (model.modelState.solution) {
			json.content.solution = model.modelState.solution.toJSON()
		}
	}
}

export default Adapter
