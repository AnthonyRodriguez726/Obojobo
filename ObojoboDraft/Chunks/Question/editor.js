import React from 'react'
import { Block } from 'slate'
import { CHILD_REQUIRED, CHILD_TYPE_INVALID, LAST_CHILD_TYPE_INVALID } from 'slate-schema-violations'

const BREAK_NODE = 'ObojoboDraft.Chunks.Break'
const CODE_NODE = 'ObojoboDraft.Chunks.Code'
const FIGURE_NODE = 'ObojoboDraft.Chunks.Figure'
const HEADING_NODE = 'ObojoboDraft.Chunks.Heading'
const IFRAME_NODE = 'ObojoboDraft.Chunks.IFrame'
const LIST_NODE = 'ObojoboDraft.Chunks.List'
const MATH_NODE = 'ObojoboDraft.Chunks.MathEquation'
const MCASSESSMENT_NODE = 'ObojoboDraft.Chunks.MCAssessment'
const QUESTION_NODE = 'ObojoboDraft.Chunks.Question'
const TABLE_NODE = 'ObojoboDraft.Chunks.Table'
const TEXT_NODE = 'ObojoboDraft.Chunks.Text'
const YOUTUBE_NODE = 'ObojoboDraft.Chunks.YouTube'

import Break from '../Break/editor'
import Code from '../Code/editor'
import Figure from '../Figure/editor'
import Heading from '../Heading/editor'
import IFrame from '../IFrame/editor'
import List from '../List/editor'
import MathEquation from '../MathEquation/editor'
import Table from '../Table/editor'
import Text from '../Text/editor'
import YouTube from '../YouTube/editor'
import MCAssessment from '../MCAssessment/editor'
import DefaultNode from '../../../src/scripts/oboeditor/default-node'

const nodes = {
	'ObojoboDraft.Chunks.Break': Break,
	'ObojoboDraft.Chunks.Code': Code,
	'ObojoboDraft.Chunks.Figure': Figure,
	'ObojoboDraft.Chunks.Heading': Heading,
	'ObojoboDraft.Chunks.IFrame': IFrame,
	'ObojoboDraft.Chunks.List': List,
	'ObojoboDraft.Chunks.MathEquation': MathEquation,
	'ObojoboDraft.Chunks.Table': Table,
	'ObojoboDraft.Chunks.Text': Text,
	'ObojoboDraft.Chunks.YouTube': YouTube,
	'ObojoboDraft.Chunks.MCAssessment': MCAssessment
}

class Node extends React.Component {
	delete() {
		const editor = this.props.editor
		const change = editor.value.change()
		change.removeNodeByKey(this.props.node.key)

		editor.onChange(change)
	}
	render(){
		return (
			<div className={'component flip-container obojobo-draft--chunks--question  is-active is-mode-practice'} {...this.props.attributes}>
				<div className={'flipper'}>
					<div className={'content back'}>
						<button className={'delete'} onClick={() => this.delete()}>X</button>
						{this.props.children}
					</div>
				</div>
			</div>
		)
	}
}

const insertNode = change => {
	change
		.insertBlock({
			type: QUESTION_NODE
		})
		.collapseToStartOfNextText()
		.focus()
}

const slateToObo = node => {
	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = node.data.get('content') || {}
	json.children = []

	node.nodes.forEach(child => {
		if(nodes.hasOwnProperty(child.type)){
			json.children.push(nodes[child.type].helpers.slateToObo(child))
		} else {
			json.children.push(DefaultNode.helpers.slateToObo(child))
		}
	})

	console.log(JSON.stringify(node.toJSON()))

	return json
}

const oboToSlate = node => {
	const json = {}
	json.object = 'block'
	json.key = node.id
	json.type = node.type
	json.data = { content: {} }

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case QUESTION_NODE:
				return <Node {...props} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.Question': {
				nodes: [
					{ types: [
						BREAK_NODE,
						CODE_NODE,
						FIGURE_NODE,
						HEADING_NODE,
						IFRAME_NODE,
						LIST_NODE,
						MATH_NODE,
						TEXT_NODE,
						TABLE_NODE,
						YOUTUBE_NODE
					], min: 1 },
					{ types: [MCASSESSMENT_NODE], min: 1, max: 1}
				],

				normalize: (change, violation, { node, child, index }) => {
					switch (violation) {
						case CHILD_REQUIRED: {
							// If we are missing the last node,
							// it should be a MCAssessemnt
							if (index === node.nodes.size){
								const block = Block.create({
									type: MCASSESSMENT_NODE,
									data: { content: { responseType: 'pick-one', shuffle: true }}
								})
								return change.insertNodeByKey(node.key, index, block)
							}

							// Otherwise, just add a text node
							const block = Block.create({
								type: TEXT_NODE,
								data: { content: { indent: 0 }}
							})
							return change.insertNodeByKey(node.key, index, block)
						}
						case CHILD_TYPE_INVALID: {
							if(child.object !== 'text') return
							return change.wrapBlockByKey(
								child.key,
								{
									type: TEXT_NODE,
									data: { content: { indent: 0 }}
								}
							)
						}
					}
				}
			}
		}
	}
}

const Question = {
	components: {
		Node,
	},
	helpers: {
		insertNode,
		slateToObo,
		oboToSlate,
	},
	plugins
}

export default Question
