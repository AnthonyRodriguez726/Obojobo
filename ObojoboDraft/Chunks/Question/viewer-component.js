import './viewer-component.scss'

let ReactCSSTransitionGroup = React.addons.CSSTransitionGroup

import Common from 'Common'
import Viewer from 'Viewer'

let { OboComponent } = Common.components
let { Dispatcher } = Common.flux
let { FocusUtil } = Common.util
let { Button } = Common.components

let { ScoreUtil } = Viewer.util
let { QuestionUtil } = Viewer.util

import QuestionContent from './Content/viewer-component'

export default class Question extends React.Component {
	onClickBlocker() {
		QuestionUtil.viewQuestion(this.props.model.get('id'))

		if (this.props.model.modelState.practice) {
			return FocusUtil.focusComponent(this.props.model.get('id'))
		}
	}

	render() {
		if (this.props.showContentOnly) {
			return this.renderContentOnly()
		}

		let score = ScoreUtil.getScoreForModel(this.props.moduleData.scoreState, this.props.model)
		let viewState = QuestionUtil.getViewState(this.props.moduleData.questionState, this.props.model)

		let assessment = this.props.model.children.models[this.props.model.children.models.length - 1]
		let AssessmentComponent = assessment.getComponentClass()

		return (
			<OboComponent
				model={this.props.model}
				moduleData={this.props.moduleData}
				className={`flip-container obojobo-draft--chunks--question${
					score === null ? '' : score === 100 ? ' is-correct' : ' is-incorrect'
				} is-${viewState}${
					this.props.model.modelState.practice ? ' is-practice' : ' is-not-practice'
				}`}
			>
				<div className="flipper">
					<div className="content back">
						<QuestionContent model={this.props.model} moduleData={this.props.moduleData} />
						<AssessmentComponent
							key={assessment.get('id')}
							model={assessment}
							moduleData={this.props.moduleData}
						/>
					</div>
					<div className="blocker front" key="blocker" onClick={this.onClickBlocker.bind(this)}>
						<Button
							value={this.props.model.modelState.practice ? 'Try Question' : 'View Question'}
						/>
					</div>
				</div>
			</OboComponent>
		)
	}

	renderContentOnly() {
		let score = ScoreUtil.getScoreForModel(this.props.moduleData.scoreState, this.props.model)
		let viewState = QuestionUtil.getViewState(this.props.moduleData.questionState, this.props.model)

		return (
			<OboComponent
				model={this.props.model}
				moduleData={this.props.moduleData}
				className={`flip-container obojobo-draft--chunks--question${
					score === null ? '' : score === 100 ? ' is-correct' : ' is-incorrect'
				} is-active${this.props.model.modelState.practice ? ' is-practice' : ' is-not-practice'}`}
			>
				<div className="flipper">
					<div className="content back">
						<QuestionContent model={this.props.model} moduleData={this.props.moduleData} />
						<div className="pad responses-hidden">(Responses Hidden)</div>
					</div>
				</div>
			</OboComponent>
		)
	}
}
