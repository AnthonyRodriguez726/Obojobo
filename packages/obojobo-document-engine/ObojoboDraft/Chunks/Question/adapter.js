import Common from 'Common'

let { OboModel } = Common.models

let Adapter = {
	construct(model, attrs) {
		if (__guard__(attrs != null ? attrs.content : undefined, x1 => x1.limit) != null) {
			model.modelState.limit = attrs.content.limit
		} else {
			model.modelState.limit = 0
		}

		if (__guard__(attrs != null ? attrs.content : undefined, x2 => x2.mode) != null) {
			model.modelState.mode = attrs.content.mode
		} else {
			model.modelState.mode = 'practice'
		}

		if (__guard__(attrs != null ? attrs.content : undefined, x3 => x3.solution) != null) {
			model.modelState.solution = OboModel.create(attrs.content.solution)
		} else {
			model.modelState.solution = null
		}
	},

	clone(model, clone) {
		clone.modelState.type = model.modelState.type
		clone.modelState.mode = model.modelState.mode
		clone.modelState.limit = model.modelState.limit
		clone.modelState.solution = null

		if (model.modelState.solution != null) {
			clone.modelState.solution = Object.assign({}, model.modelState.solution)
		}
	},

	toJSON(model, json) {
		json.content.type = model.modelState.type
		json.content.solution = null

		if (model.modelState.solution != null) {
			json.content.solution = model.modelState.solution.toJSON()
		}
	}
}

export default Adapter
function __guard__(value, transform) {
	return typeof value !== 'undefined' && value !== null ? transform(value) : undefined
}
