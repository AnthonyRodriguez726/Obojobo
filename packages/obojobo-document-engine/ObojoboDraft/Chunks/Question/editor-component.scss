@import '../../../src/scss/includes';

.obojobo-draft--chunks--question {
	> .flipper {
		&.question-editor {
			background: $color-bg;
		}
	}

	.add-solution {
		left: 50%;
		transform: translate(-50%, 0);
		position: relative;
		margin: $dimension-padding / 3;
	}
}
