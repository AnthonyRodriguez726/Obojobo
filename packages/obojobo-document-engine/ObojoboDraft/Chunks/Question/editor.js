import React from 'react'
import { Block } from 'slate'
import { CHILD_REQUIRED, CHILD_TYPE_INVALID } from 'slate-schema-violations'
import Common from 'Common'

const MCASSESSMENT_NODE = 'ObojoboDraft.Chunks.MCAssessment'
const QUESTION_NODE = 'ObojoboDraft.Chunks.Question'
const TEXT_NODE = 'ObojoboDraft.Chunks.Text'
const SOLUTION_NODE = 'ObojoboDraft.Chunks.Question.Solution'
const PAGE_NODE = 'ObojoboDraft.Pages.Page'

import Page from '../../Pages/Page/editor'
import Component from '../../../src/scripts/oboeditor/components/editor-component'

import emptyNode from './empty-node.json'
import Icon from './icon'

const { Button } = Common.components

const Solution = props => {
	const deleteNode = () => {
		const editor = props.editor
		const change = editor.value.change()
		change.removeNodeByKey(props.node.key)

		editor.onChange(change)
	}

	return (
		<div className={'solution-editor'}>
			{props.children}
			<button className={'delete-node'} onClick={() => deleteNode()}>
				X
			</button>
		</div>
	)
}

class Node extends React.Component {
	delete() {
		const editor = this.props.editor
		const change = editor.value.change()
		change.removeNodeByKey(this.props.node.key)

		editor.onChange(change)
	}
	addSolution() {
		const editor = this.props.editor
		const change = editor.value.change()

		const newQuestion = Block.create({
			type: SOLUTION_NODE
		})
		change.insertNodeByKey(this.props.node.key, this.props.node.nodes.size, newQuestion)

		editor.onChange(change)
	}
	render() {
		const hasSolution = this.props.node.nodes.last().type === SOLUTION_NODE
		return (
			<div className={'component obojobo-draft--chunks--question is-viewed is-mode-practice pad'}>
				<div className={'flipper question-editor'}>
					<div className={'content-back'}>
						{this.props.children}
						{hasSolution ? null : (
							<Button className={'add-solution'} onClick={() => this.addSolution()}>
								{'Add Solution'}
							</Button>
						)}
					</div>
				</div>
				<button className={'delete-node'} onClick={() => this.delete()}>
					X
				</button>
			</div>
		)
	}
}

const insertNode = change => {
	change
		.insertBlock(Block.fromJSON(emptyNode))
		.focus()
}

const slateToObo = node => {
	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = node.data.get('content') || {}
	json.children = []

	delete json.content.solution

	node.nodes.forEach(child => {
		if (child.type === SOLUTION_NODE) {
			json.content.solution = Page.helpers.slateToObo(child.nodes.get(0))
		} else if (child.type === MCASSESSMENT_NODE) {
			json.children.push(Common.Store.getItemForType(child.type).slateToObo(child))
		} else {
			json.children.push(Component.helpers.slateToObo(child))
		}
	})

	return json
}

const oboToSlate = node => {
	const json = {}
	json.object = 'block'
	json.key = node.id
	json.type = node.type
	json.data = { content: node.content }
	json.nodes = []

	node.children.forEach(child => {
		if(child.type === MCASSESSMENT_NODE) {
			json.nodes.push(Common.Store.getItemForType(child.type).oboToSlate(child))
		}
		json.nodes.push(Component.helpers.oboToSlate(child))
	})

	if (json.data.content.solution) {
		const solution = {
			object: 'block',
			type: SOLUTION_NODE,
			nodes: []
		}

		solution.nodes.push(Page.helpers.oboToSlate(json.data.content.solution))
		json.nodes.push(solution)
	}

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case QUESTION_NODE:
				return <Node {...props} {...props.attributes} />
			case SOLUTION_NODE:
				return <Solution {...props} {...props.attributes} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.Question': {
				nodes: [
					{ match: [ { type: 'oboeditor.component' } ], min: 1 },
					{ match: [MCASSESSMENT_NODE], min: 1, max: 1 },
					{ match: [SOLUTION_NODE], max: 1 }
				],

				normalize: (change, error) => {
					const { node, child, index } = error
					switch (error.code) {
						case CHILD_REQUIRED: {
							// If we are missing the last node,
							// it should be a MCAssessment
							if (index === node.nodes.size) {
								const block = Block.create({
									type: MCASSESSMENT_NODE,
									data: { content: { responseType: 'pick-one', shuffle: true } }
								})
								return change.insertNodeByKey(node.key, index, block)
							}

							// Otherwise, just add a text node
							const block = Block.create({
								object: 'block',
								type: 'oboeditor.component',
								nodes: [
									{
										object: 'block',
										type: TEXT_NODE
									}
								]
							})
							return change.insertNodeByKey(node.key, index, block)
						}
						case CHILD_TYPE_INVALID: {
							if (child.object !== 'text') return
							const block = Block.fromJSON({
								object: 'block',
								type: 'oboeditor.component',
								nodes: [
									{
										object: 'block',
										type: TEXT_NODE
									}
								]
							})
							return change.withoutNormalization(c => {
								c.removeNodeByKey(child.key)
								return c.insertNodeByKey(node.key, index, block)
							})
						}
					}
				}
			},
			'ObojoboDraft.Chunks.Question.Solution': {
				nodes: [
					{
						match: [{ type: PAGE_NODE }],
						min: 1,
						max: 1
					}
				],
				normalize: (change, error) => {
					const { node, child, index } = error
					switch (error.code) {
						case CHILD_REQUIRED: {
							const block = Block.create({
								type: PAGE_NODE
							})
							return change.insertNodeByKey(node.key, index, block)
						}
						case CHILD_TYPE_INVALID: {
							return change.wrapBlockByKey(child.key, {
								type: PAGE_NODE
							})
						}
					}
				}
			}
		}
	}
}

const Question = {
	name: QUESTION_NODE,
	components: {
		Node,
		Solution,
		Icon
	},
	helpers: {
		insertNode,
		slateToObo,
		oboToSlate
	},
	json: {
		emptyNode
	},
	plugins
}

Common.Store.registerEditorModel('ObojoboDraft.Chunks.Question', {
	name: 'Question',
	icon: Icon,
	isInsertable: true,
	insertJSON: emptyNode,
	slateToObo,
	oboToSlate,
	plugins
})

export default Question
