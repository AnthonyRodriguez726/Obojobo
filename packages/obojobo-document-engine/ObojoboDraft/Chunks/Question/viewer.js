import Common from 'Common'
import adapter from './adapter'
import ViewerComponent from './viewer-component'

let SelectionHandler = Common.chunk.textChunk.TextGroupSelectionHandler

Common.Store.registerModel('ObojoboDraft.Chunks.Question', {
	type: 'chunk',
	adapter: adapter,
	componentClass: ViewerComponent,
	selectionHandler: new SelectionHandler(),
	getNavItem(model) {
		let label
		let questions = model.parent.children.models.filter(
			child => child.get('type') === 'ObojoboDraft.Chunks.Question'
		)

		if (model.title) {
			label = model.title
		} else if (model.modelState.mode === 'practice') {
			label = `Practice Question ${questions.indexOf(model) + 1}`
		} else {
			label = `Question ${questions.indexOf(model) + 1}`
		}

		return {
			type: 'sub-link',
			label,
			path: [`#obo-${model.get('id')}`]
		}
	}
})
