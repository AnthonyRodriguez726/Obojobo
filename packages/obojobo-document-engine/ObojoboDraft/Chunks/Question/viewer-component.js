import './viewer-component.scss'

import React from 'react'

import Common from 'Common'
import Viewer from 'Viewer'
import isOrNot from '../../../src/scripts/common/isornot'

const { OboComponent } = Common.components
const { FocusUtil } = Common.util
const { Button } = Common.components

const { QuestionUtil } = Viewer.util

import QuestionContent from './Content/viewer-component'

export default class Question extends React.Component {
	onClickBlocker() {
		QuestionUtil.viewQuestion(this.props.model.get('id'))
		const mode = this.props.mode ? this.props.mode : this.props.model.modelState.mode

		FocusUtil.focusComponent(this.props.model.get('id'), mode === 'practice')
	}

	render() {
		if (this.props.showContentOnly) {
			return this.renderContentOnly()
		}

		const score = QuestionUtil.getScoreForModel(
			this.props.moduleData.questionState,
			this.props.model,
			this.props.moduleData.navState.context
		)
		const viewState = QuestionUtil.getViewState(
			this.props.moduleData.questionState,
			this.props.model
		)

		const assessment = this.props.model.children.models[this.props.model.children.models.length - 1]
		const AssessmentComponent = assessment.getComponentClass()

		const mode = this.props.mode ? this.props.mode : this.props.model.modelState.mode

		let scoreClassName
		switch (score) {
			case null:
				scoreClassName = ''
				break

			case 100:
				scoreClassName = ' is-correct'
				break

			default:
				scoreClassName = ' is-not-correct'
				break
		}

		const classNames =
			'obojobo-draft--chunks--question' +
			scoreClassName +
			(mode === 'review' ? ' is-active' : ` is-${viewState}`) +
			` is-mode-${mode}`

		return (
			<OboComponent
				model={this.props.model}
				moduleData={this.props.moduleData}
				className={classNames}
				tabIndex="-1"
			>
				<div className="flipper">
					<div className="content-back">
						<QuestionContent model={this.props.model} moduleData={this.props.moduleData} />
						<AssessmentComponent
							key={assessment.get('id')}
							model={assessment}
							moduleData={this.props.moduleData}
							mode={mode}
						/>
					</div>
					<div className="blocker-front" key="blocker" onClick={this.onClickBlocker.bind(this)}>
						<Button value={mode === 'practice' ? 'Try Question' : 'View Question'} />
					</div>
				</div>
			</OboComponent>
		)
	}

	renderContentOnly() {
		const score = QuestionUtil.getScoreForModel(
			this.props.moduleData.questionState,
			this.props.model,
			this.props.moduleData.navState.context
		)

		const mode = this.props.mode ? this.props.mode : this.props.model.modelState.mode

		const className =
			'obojobo-draft--chunks--question' +
			isOrNot(score === 100, 'correct') +
			' is-active' +
			` is-mode-${mode}`

		return (
			<OboComponent
				model={this.props.model}
				moduleData={this.props.moduleData}
				className={className}
				tabIndex="-1"
			>
				<div className="flipper">
					<div className="content back">
						<QuestionContent model={this.props.model} moduleData={this.props.moduleData} />
						<div className="pad responses-hidden">(Responses Hidden)</div>
					</div>
				</div>
			</OboComponent>
		)
	}
}
