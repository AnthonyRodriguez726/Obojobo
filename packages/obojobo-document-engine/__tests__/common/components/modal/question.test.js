import { shallow } from 'enzyme'
import React from 'react'
import Question from '../../../../src/scripts/common/components/modal/question'
import renderer from 'react-test-renderer'

test('Question', () => {
	const component = renderer.create(
		<Question
			cancel="cancel"
			reject="reject"
			confirm="confirm"
			rejectButtonLabel="Reject Label"
			confirmButtonLabel="Confirm Label"
			cancelOnReject={true}
			modal={{
				onButtonClick: jest.fn()
			}}
		>
			Content
		</Question>
	)
	let tree = component.toJSON()

	expect(tree).toMatchSnapshot()
})

test('Question confirm click', () => {
	let onClick = jest.fn()

	const component = shallow(
		<Question
			cancel="cancel"
			reject="reject"
			confirm="confirm"
			rejectButtonLabel="Reject Label"
			confirmButtonLabel="Confirm Label"
			cancelOnReject={true}
			modal={{
				onButtonClick: onClick
			}}
		>
			Content
		</Question>
	)

	let rejectButton = component.find('button').at(0)
	let confirmButton = component.find('button').at(1)

	confirmButton.simulate('click')

	expect(onClick).toHaveBeenCalledWith('confirm')
})

test('Question cancel click', () => {
	let onClick = jest.fn()

	const component = shallow(
		<Question
			cancel="cancel"
			reject="reject"
			confirm="confirm"
			rejectButtonLabel="Reject Label"
			confirmButtonLabel="Confirm Label"
			cancelOnReject={true}
			modal={{
				onButtonClick: onClick
			}}
		>
			Content
		</Question>
	)

	let rejectButton = component.find('button').at(0)
	let confirmButton = component.find('button').at(1)

	rejectButton.simulate('click')

	expect(onClick).toHaveBeenCalledWith('cancel')
})

test('Question reject click', () => {
	let onClick = jest.fn()

	const component = shallow(
		<Question
			cancel="cancel"
			reject="reject"
			confirm="confirm"
			rejectButtonLabel="Reject Label"
			confirmButtonLabel="Confirm Label"
			cancelOnReject={false}
			modal={{
				onButtonClick: onClick
			}}
		>
			Content
		</Question>
	)

	let rejectButton = component.find('button').at(0)
	let confirmButton = component.find('button').at(1)

	rejectButton.simulate('click')

	expect(onClick).toHaveBeenCalledWith('reject')
})
