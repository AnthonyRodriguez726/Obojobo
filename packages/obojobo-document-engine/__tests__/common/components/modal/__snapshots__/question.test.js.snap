// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question 1`] = `
<div>
  <p>
    Content
  </p>
  <button
    onClick={[Function]}
  >
    Reject Label
  </button>
  <button
    onClick={[Function]}
  >
    Confirm Label
  </button>
</div>
`;
